source/tutorials/write-ops.txt

```diff
@@ -0,0 +1,369 @@
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, insert data, change data
+
+.. toctree::
+   
+   /tutorials/bulk-writes/
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+You can perform write operations to insert new documents,
+update existing documents, replace an existing document, or delete
+existing documents from a collection.
+
+Prerequisites
+-------------
+
+.. include:: /includes/prereq-restaurants.rst
+
+.. code-block:: scala
+
+   import org.mongodb.scala._
+   import org.mongodb.scala.model._
+   import org.mongodb.scala.model.Filters._
+   import org.mongodb.scala.model.Updates._
+   import org.mongodb.scala.model.UpdateOptions
+   import org.mongodb.scala.bson.BsonObjectId
+
+.. include:: /includes/obs-note.rst
+
+Connect to a MongoDB Deployment
+-------------------------------
+
+.. include:: /includes/connect-section.rst
+
+Insert a Document
+-----------------
+
+To insert a single document into the collection, you can use the
+collection's ``insertOne()`` method:
+
+.. code-block:: scala
+
+   val document = Document("name" -> "Café Con Leche" , 
+                           "contact" -> Document("phone" -> "228-555-0149", 
+                                                 "email" -> "cafeconleche@example.com", 
+                                                 "location" -> Seq(-73.92502, 40.8279556)),
+                           "stars" -> 3, "categories" -> Seq("Bakery", "Coffee", "Pastries"))
+   
+   collection.insertOne(document).printResults()
+
+.. note::
+
+   If no top-level ``id`` field is specified in the document, MongoDB
+   automatically generates a value and adds this field to the inserted
+   document.
+
+Insert Multiple Documents
+-------------------------
+
+To insert multiple documents, you can use the collection's
+``insertMany()`` method, which takes a list of documents to insert as a parameter.
+
+The following example inserts two documents into the collection:
+
+.. code-block:: scala
+
+   val doc1 = Document("name" -> "Amarcord Pizzeria" , 
+                       "contact" -> Document("phone" -> "264-555-0193", 
+                                             "email" -> "amarcord.pizzeria@example.net", 
+                                             "location" -> Seq(-73.88502, 40.749556)),
+                       "stars" -> 2, "categories" -> Seq("Pizzeria", "Italian", "Pasta"))
+   
+   val doc2 = Document("name" -> "Blue Coffee Bar" , 
+                       "contact" -> Document("phone" -> "604-555-0102", 
+                                             "email" -> "bluecoffeebar@example.com", 
+                                             "location" -> Seq(-73.97902, 40.8479556)),
+                       "stars" -> 5, "categories" -> Seq("Coffee", "Pastries"))
+   
+   collection.insertMany(Seq(doc1, doc2)).printResults()
+
+.. note::
+
+   If no top-level ``id`` field is specified in the document, MongoDB
```

Comment:
Suggestion:

I think this should either be the field name "_id", or unformatted text "id" to refer to the identifier.

```suggestion
   If no top-level ``_id`` field is specified in the document, MongoDB
```